src/Microsoft.ML.Data/Commands/TrainCommand.cs

```diff
@@ -223,19 +223,17 @@ public static string MatchNameOrDefaultOrNull(IExceptionContext ectx, ISchema sc
         public static IPredictor Train(IHostEnvironment env, IChannel ch, RoleMappedData data, ITrainer trainer,
             ICalibratorTrainerFactory calibrator, int maxCalibrationExamples)
```

Comment:
>ICalibratorTrainerFactory [](start = 12, length = 25)

Can we decide what interface are we exposing to user? this one or IComponentFactory<ICalibratorTrainer>?

Reply:
It looks like issue was exactly about it, but you changed private method instead of public.

---
In reply to: [222828179](https://github.com/dotnet/machinelearning/pull/1155#discussion_r222828179) [](ancestors = 222828179)

Reply:
Sorry I thought I had made that change as it should be IComponentFactory<ICalibratorTrainer>. I will update the public interface.

---
In reply to: [222828755](https://github.com/dotnet/machinelearning/pull/1155#discussion_r222828755) [](ancestors = 222828755,222828179)